### DISCUSSION TITLE
Getting Flux v2 compatible with v1.15

### DISCUSSION THREAD

**Discussion by dieend** [fluxcd/flux2#905] [Q&A] [answered]
Hi, I know this has been asked here.
https://github.com/fluxcd/flux2/issues/539

I still tried to convert the resources generated by `flux install --export --watch-all-namespaces false --namespace test-flux` and make it compatible with 1.15.

I was able to install it, but the controller encountered some errors
`test-flux/helm-controller-6497659cf8-w68rg`
```
E0212 04:36:36.835309       6 leaderelection.go:325] error retrieving resource lock pre-12345/5b6ca942.fluxcd.io: configmaps "5b6ca942.fluxcd.io" is forbidden: User "system:serviceaccount:pre-12345:helm-controller" cannot get resource "configmaps" in API group "" in the namespace "pre-12345"
```
`test-flux/source-controller-67b9c66f9f-j8f8m/manager`
```
 E0212 08:48:19.812476       6 reflector.go:138] pkg/mod/k8s.io/client-go@v0.20.2/tools/cache/reflector.go:167: Failed to watch *v1beta1.HelmRepository: failed to list *v1beta1.HelmRepository: helmrepositories.source.toolkit.fluxcd.io is forbidden: User
  "system:serviceaccount:pre-12345:source-controller" cannot list resource "helmrepositories" in API group "source.toolkit.fluxcd.io" at the cluster scope
 E0212 08:48:20.712689       6 leaderelection.go:325] error retrieving resource lock pre-12345/305740c0.fluxcd.io: configmaps "305740c0.fluxcd.io" is forbidden: User "system:serviceaccount:pre-12345:source-controller" cannot get resource "configmaps" in
  API group "" in the namespace "pre-12345"
  ```
  
  Is it truly impossible to get this running in v1.15 cluster?

**Comment by stefanprodan** [comment, marked as answer]
Judging by the RBAC errors you are not using the latest flux CLI.

**Reply by dieend** [reply]
Thanks! You are correct. I was able to get all the controllers running in our 1.15 Kubernetes cluster.

**Comment by hiddeco** [comment]
> Is it truly impossible to get this running in v1.15 cluster?

It is not _supported_ by us, you may still get it to work with patches but by doing this you are automatically on your own on in terms of support. Plus if you get it to work, it may break at any moment in time as we do not take those versions into account during development.

**Reply by dieend** [reply]
Understood. We are in the prototyping phase so it shouldn't be an issue.
